docs/development.rst

```diff
@@ -0,0 +1,160 @@
+=====================
+Development Standards
+=====================
+
+Front End Development
+=====================
+
+Background
+----------
+
+.. info::
+
+    Consider this the cononical resource for contributing Javascript and CSS. We
+    are currently in the process of modernizing our front end development
+    procedures. You will see a lot of different styles around the code base for
+    front end JavaScript and CSS.
+
+Our modern front end development stack includes the following tools:
+
+* `Gulp`_
+* `Bower`_
+* `Browserify`_
+* `Debowerify`_
```

Comment:
I have the feeling the whole JS front end development moves completely over to npm. It might make sense to completely drop bower and only use NPM. I think all current dependencies are already available in npm, except the readthedocs-client. But we can install it directly from github as `npm install` supports installing from git and tarballs: https://docs.npmjs.com/cli/install